tests/test_utils.py

```diff
@@ -9,43 +9,51 @@
 import adios4dolfinx
 
 
-def write_function(mesh, el, f, dtype) -> str:
+def write_function(mesh, el, f, dtype, name="uh", append: bool = False) -> str:
     V = dolfinx.fem.functionspace(mesh, el)
     uh = dolfinx.fem.Function(V, dtype=dtype)
     uh.interpolate(f)
+    uh.name = name
     el_hash = (
         V.element.signature()
         .replace(" ", "")
         .replace(",", "")
         .replace("(", "")
         .replace(")", "")
+        .replace("[", "")
+        .replace("]", "")
     )
-    filename = pathlib.Path(f"output/mesh{el_hash}_{dtype}.bp")
-    if mesh.comm.size != 1:
-        adios4dolfinx.write_mesh(mesh, filename)
-        adios4dolfinx.write_function(uh, filename)
 
+    file_hash = f"{el_hash}_{np.dtype(dtype).name}"
+    filename = pathlib.Path(f"output/mesh_{file_hash}.bp")
+    if mesh.comm.size != 1:
```

Comment:
I am confused by the logic here. It seems to me that it is saying
```
if running in parallel:
   call write free functions from every process
else:  # running in serial
   if on process 0:  # which is obviously true in serial
     call write free functions on the root process
```
which basically is equivalent to
```
call write free functions from every process
```

Reply:
There is a subtle thing here. `mesh.comm`!=MPI.COMM_WORLD.
In the case above, the mesh can live on COMM_WORLD or COMM_SELF.
If the mesh lives on COMM_SELF, I only store the one that is on rank 0 (as the filename does not depend on rank).
This is to be able to test: write_serial/read_parallel and write_parallel/read_serial within the same code.

Reply:
OK thanks, I hadn't noticed that the two communicators were different. I will resume my testing tomorrow morning.

Reply:
Maybe wait until I merge #51 as it will make your life faster?:) i aim to merge it within an hour.